src/main/java/cs361/battleships/models/Ship.java

```diff
@@ -25,28 +26,63 @@ public Ship(String kind) {
 		this.kind = kind;
 		switch(kind) {
 			case "MINESWEEPER":
+				this.underwater = false;
 				size = 2;
 				break;
 			case "DESTROYER":
+				this.underwater = false;
 				size = 3;
 				break;
 			case "BATTLESHIP":
+				this.underwater = false;
 				size = 4;
 				break;
+			case "SUB":
+				this.underwater = false;
+				size = 5;
+				break;
+			case "SUBB":
+				this.underwater = true;
+				this.kind = "SUB";
+				size = 5;
+				break;
 		}
 	}
 
+	public boolean isUnderwater(){
+		return underwater;
+	}
+	public void goUnderwater() { this.underwater = true;}
+
 	public List<Square> getOccupiedSquares() {
 		return occupiedSquares;
 	}
 
 	public void place(char col, int row, boolean isVertical) {
-		for (int i=0; i<size; i++) {
-			if (isVertical) {
-				occupiedSquares.add(new Square(row+i, col));
-			} else {
-				occupiedSquares.add(new Square(row, (char) (col + i)));
-			}
+		switch (this.kind) {
+			case "SUB":
+				for (int i = 0; i < size - 1; i++) {
+					if (isVertical) {
+						occupiedSquares.add(new Square(row + i, col));
+					} else {
+						occupiedSquares.add(new Square(row, (char) (col + i)));
+					}
+				}
+				if (isVertical) {
+					occupiedSquares.add(new Square(row + 1, (char) (col + 1)));
+				} else {
+					occupiedSquares.add(new Square(row - 1, (char) (col + 1)));
+				}
+				break;
+			default:
+				for (int i = 0; i < size; i++) {
+					if (isVertical) {
+						occupiedSquares.add(new Square(row + i, col));
+					} else {
+						occupiedSquares.add(new Square(row, (char) (col + i)));
+					}
+				}
+				break;
```

Comment:
This is important.  Handles the new shape of the sub well while also keeping functionality for other ship types.  Good job.